Fix MemAlign bug bash issues (#20712)
mlflow/mlflow: docs/docs/genai/eval-monitor/scorers/llm-judge/memalign.mdx

MemAlign requires additional dependencies:

```bash

> pip install mlflow[genai] litellm dspy jinja2 tqdm

Reviewer: `mlflow[genai]` should already include litellm

)

# Retrieve traces with human feedback
all_traces = mlflow.search_traces(return_type="list")
alignment_traces = [
    trace for trace in all_traces
    if any(
        feedback.name == "politeness" for feedback in trace.info.assessments
        # feedback name must match the judge name for alignment to work
    )
]

# Align the judge
aligned_judge = judge.align(traces=alignment_traces, optimizer=optimizer)
```

## Parameters

| Parameter         | Type  | Default                            | Description                                                                                                                                                                        |
| ----------------- | ----- | ---------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `reflection_lm`   | `str` | See below                          | Model used for extracting guidelines from feedback. See [Supported Judge Models](/genai/eval-monitor/scorers/llm-judge/custom-judges/supported-models) section for how to specify. |

Reviewer 2: ```suggestion
| `reflection_lm`   | `str` | `"openai:/gpt-4o-mini"`                          | Model used for extracting guidelines from feedback. Must be in `<provider>:/<model-name>` format. |
```

Author: the default should depend on tracking uri (`databricks` when in databricks, and `openai:/gpt-4o-mini` otherwise), right? 
i do hope to link the supported judge models page in case a user lands on the memalign page first, and is unfamiliar with litellm

| `retrieval_k`     | `int` | `5`                                | Number of relevant examples to retrieve from episodic memory during inference.                                                                                                     |
| `embedding_model` | `str` | `"openai:/text-embedding-3-small"` | Model for episodic memory retrieval. Must be in `<provider>:/<model-name>` format.                                                                                                 |

The number of parallel threads for LLM calls during guideline distillation can be configured via the `MLFLOW_GENAI_OPTIMIZE_MAX_WORKERS` environment variable (default: 8).

> The default `reflection_lm` depends on the tracking URI setup:

Author: sg. The supported judge models page doesn't talk about default models directly, tho. i'll add a section there
